pkg/controller/compliancescan/scan.go

```diff
@@ -219,14 +249,192 @@ func newScanPodForNode(scanInstance *compv1alpha1.ComplianceScan, node *corev1.N
 	}
 }
 
+func newPlatformScanPod(scanInstance *compv1alpha1.ComplianceScan, logger logr.Logger) *corev1.Pod {
+	mode := int32(0744)
+	podName := getPodForNodeName(scanInstance.Name, PlatformScanName)
+	cmName := getConfigMapForNodeName(scanInstance.Name, PlatformScanName)
+	podLabels := map[string]string{
+		"complianceScan": scanInstance.Name,
+	}
+	collectorArgs := []string{
+		"--content=/content/" + scanInstance.Spec.Content,
+		"--resultdir=" + PlatformScanDataRoot,
+		"--profile=" + scanInstance.Spec.Profile,
+	}
+
+	if scanInstance.Spec.Debug {
+		collectorArgs = append(collectorArgs, "--debug")
+	}
+
+	return &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: scanInstance.Namespace,
+			Labels:    podLabels,
+		},
+		Spec: corev1.PodSpec{
+			ServiceAccountName: resultscollectorSA,
```

Comment:
Wouldn't it be better to create a separate role just for the API collector rather than reuse two roles for the same thing?

Reply:
Yep, that's a follow-up